Tools/clinic/clinic.py

```diff
@@ -1954,27 +1954,32 @@ def dump(self):
         return "".join(texts)
 
 
+@dc.dataclass(slots=True, repr=False)
```

Comment:
```suggestion
@dc.dataclass(slots=True, frozen=True, repr=False)
```

Looks like, it is not ever mutated.

Reply:
This would mean we'd have to use the `object.__setattr__` hack in `__post_init__` for setting `filename`, right? That wouldn't be a _huge_ deal, but I also don't really see much _advantage_ to making this class frozen

Reply:
It is clearly up to you!

I prefer immutable data structures in my projects everywhere, but I don't impose my views on others :)

Reply:
In general I agree, but here I'm inclined to think it doesn't matter too much, so I'd rather keep the code in `__post_init__` as readable as possible :)

Thanks for the review!